src/item.cpp

```diff
@@ -4621,16 +4623,55 @@ std::string item::tname( unsigned int quantity, bool with_prefix, unsigned int t
+                }
+                colorsuffix = "</color>";
+            }
+
             contents_suffix_text = string_format( pgettext( "item name",
                                                   //~ [container item name] " > [inner item  name]"
-                                                  " > %1$s" ),
+                                                  " > %1$s%2$s" ),
+                                                  /* with_contents=false for nested items to prevent excessively long names */
+                                                  colorprefix, contents_item.tname( contents_count, true, 0, false )
                                                   /* with_contents=false for nested items to prevent excessively long names */
-                                                  contents_item.tname( contents_count, true, 0, false ) );
+                                                ) + string_format( "%s", colorsuffix );
+
         } else if( !contents.empty() ) {
-            contents_suffix_text = string_format( npgettext( "item name",
+
+            int worstrot = 0;
+                // If container has multiple food items, take the worst rot state and use that as the color for the "# items" text.
+                // 0 = No food; 1 = Non-perishable (cyan); 2 = Perishable (light cyan); 3 = Old (yellow); 4 = Rotten (brown)
+                for( const item *it : contents.all_items_top( item_pocket::pocket_type::CONTAINER ) ) {
+                    if( worstrot < 4 && it->rotten() ) {
+                        worstrot = 4;
+                        colorprefix = "<color_brown>";
+                    } else if( worstrot < 3 && it->is_going_bad() ) {
+                        worstrot = 3;
+                        colorprefix = "<color_yellow>";
+                    } else if( worstrot < 2 && it->goes_bad() ) {
+                        worstrot = 2;
+                        colorprefix = "<color_light_cyan>";
+                    } else if( worstrot < 1 && it->is_food() ) {
+                        worstrot = 1;
+                        colorprefix = "<color_cyan>";
+                    }
+                }
+            
+            colorsuffix = ( worstrot > 0 ) ? "</color>" : "";
+            contents_suffix_text = string_format( pgettext( "item name",
                                                   //~ [container item name] " > [count] item"
-                                                  " > %1$zd item", " > %1$zd items",
-                                                  contents.num_item_stacks() ), contents.num_item_stacks() );
+                                                  " > %1$s" ), colorprefix ) +
+                                   string_format( ngettext( "%1$zd item", "%1$zd items", contents.num_item_stacks() ),
+                                                  contents.num_item_stacks() ) +
+                                   string_format( pgettext( "color suffix", "%1$s" ), colorsuffix );
```

Comment:
Please store the color instead of the color prefix/suffix strings and use `colorize( text, color )` (from color.h) to add color to the text.

Reply:
(Oh huh, that's odd, didn't realize this comment was duplicated; addressed it in the latest commit!)